pseudomonas_analysis/2_identify_generic_genes_pathways_pao1.ipynb

```diff
@@ -0,0 +1,1764 @@
+    "\n",
+    "    Returns\n",
+    "    --------\n",
+    "    simulated_data_file: str\n",
+    "        File containing simulated gene expression data\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # Files\n",
+    "    NN_dir = os.path.join(base_dir, dataset_name, \"models\", NN_architecture)\n",
+    "    latent_dim = NN_architecture.split(\"_\")[-2] # Here we manually changed the index to work\n",
+    "\n",
+    "    model_encoder_file = glob.glob(os.path.join(NN_dir, \"*_encoder_model.h5\"))[0]\n",
+    "\n",
+    "    weights_encoder_file = glob.glob(os.path.join(NN_dir, \"*_encoder_weights.h5\"))[0]\n",
+    "\n",
+    "    model_decoder_file = glob.glob(os.path.join(NN_dir, \"*_decoder_model.h5\"))[0]\n",
+    "\n",
+    "    weights_decoder_file = glob.glob(os.path.join(NN_dir, \"*_decoder_weights.h5\"))[0]\n",
+    "    \n",
+    "    #print(\"model files used\")\n",
+    "    #print(model_encoder_file)\n",
+    "\n",
+    "    # Load saved models\n",
+    "    loaded_model = load_model(model_encoder_file, compile=False)\n",
+    "    loaded_decode_model = load_model(model_decoder_file, compile=False)\n",
+    "\n",
+    "    loaded_model.load_weights(weights_encoder_file)\n",
+    "    loaded_decode_model.load_weights(weights_decoder_file)\n",
+    "\n",
+    "    # Read data\n",
+    "    normalized_data = pd.read_csv(normalized_data_file, header=0, sep=\"\\t\", index_col=0)\n",
+    "    #print(normalized_data_file)\n",
+    "\n",
+    "    # Get corresponding sample ids\n",
+    "    sample_ids = get_sample_ids(selected_experiment_id, dataset_name, sample_id_colname)\n",
+    "\n",
+    "    # Gene expression data for selected samples\n",
+    "    selected_data_df = normalized_data.loc[sample_ids]\n",
+    "\n",
+    "    # Encode selected experiment into latent space\n",
+    "    data_encoded = loaded_model.predict_on_batch(selected_data_df)\n",
+    "    data_encoded_df = pd.DataFrame(data_encoded, index=selected_data_df.index)\n",
+    "\n",
+    "    # Get centroid of original data\n",
+    "    centroid = data_encoded_df.mean(axis=0)\n",
+    "\n",
+    "    # Add individual vectors(centroid, sample point) to new_centroid\n",
+    "\n",
+    "    # Encode original gene expression data into latent space\n",
+    "    data_encoded_all = loaded_model.predict_on_batch(normalized_data)\n",
+    "    data_encoded_all_df = pd.DataFrame(data_encoded_all, index=normalized_data.index)\n",
+    "\n",
+    "    data_encoded_all_df.head()\n",
+    "\n",
+    "    # Find a new location in the latent space by sampling from the latent space\n",
+    "    encoded_means = data_encoded_all_df.mean(axis=0)\n",
+    "    encoded_stds = data_encoded_all_df.std(axis=0)\n",
+    "\n",
+    "    latent_dim = int(latent_dim)\n",
+    "    new_centroid = np.zeros(latent_dim)\n",
+    "\n",
+    "    for j in range(latent_dim):\n",
+    "        new_centroid[j] = np.random.normal(encoded_means[j], encoded_stds[j])\n",
```

Comment:
Why is the new centroid a random variable rather than being set to the centroid of the template data?

Reply:
So the idea behind this simulation is that it takes a template experiment (this is your real gene expression experiment) and linearly shifts this template experiment to a new random location (where the new location is based on the latent space distribution). This shifted template experiment becomes your simulated experiment. So the shift vector is connecting the centroid of the template experiment to the random new location (new_centroid). Hopefully that makes more sense.